packages/react-server/core/renderMiddleware.js

```diff
@@ -295,6 +296,13 @@ function setDataBundleContentType(req, res) {
 	res.set('Content-Type', 'application/json');
 }
 
+function setHeaders(req, res, context, start, pageObject) {
```

Comment:
Yeah, I think `setResponseHeaders` or `setHttpHeaders` would be clearer, since we already have `writeHeader`, referring to `<head>`